### ISSUE TITLE
Update the variant syntax

### ISSUE THREAD

**Issue by colmtuite** [modulz/stitches#85]
After chatting with @peduarte, we agreed the variant syntax could be improved. After testing the lib, I think it gets very difficult to scan when you have lots of variants.

We agreed on this syntax:

```
const Button = styled(element, {
  appearance: 'none'
},
{
  variant: {
    // styles
  }
},
{
  size: {
    // styles
  }
})
```

However, myself and @StephenHaney still think totally separate variant functions is the best option.

**Comment by peduarte** [comment]
Yep, we could support this approach. It's similar to how Styled Components function can receive many comma-separated style objects.

Should we continue to support multiple variants in one object? I don't see any harm in doing so.

**Comment by christianalfoni** [comment]
I am waiting with this for a decision. I totally see the "scanning argument". I would argue it feels a bit unnatural to split up into multiple objects where you are only allowed to have one key. But if allowing multiple keys, the reason for multiple arguments looses its purpose.

So this is like a syntax discussion, which is scary in itself 😂 But you can improve scanning by adding quotes for example:

```ts
const Button = styled(element, {
  appearance: 'none'
},
{
  "variant": {
    // styles
  },
  "size": {
    // styles
  }
}
```

Or make the values constants:

```ts
const Button = styled(element, {
  appearance: 'none'
},
{
  variant: {
    MUTED: {}
  },
  size: {
    SMALL: {},
    LARGE: {}
  }
}
```

It seems that the multiple arguments is a compromise? How does the separate variants function look like? 😄

**Comment by peduarte** [comment]
@christianalfoni we had considered separate functions as well, something like:
```jsx
const Button = styled();

Button.variant('size', {
  small: {},
  large: {}
})
```

But im worried about increasing too much the API surface area. Maybe not a problem. Whats your opinion on this? I know that Colm likes it.

**Comment by colmtuite** [comment]
@peduarte in your recent tweet I saw you had line breaks between the variants. Even that helps a lot I think. So maybe it is just a simple syntax preference thing, not sure.

I think @StephenHaney likes the separate functions approach too?

**Comment by StephenHaney** [comment]
I do, all else being equal, but I do acknowledge it's likely more complicated to implement. If we do go with a function call per variant, I like Pedro's idea of supporting both syntaxes as well.

**Comment by jjenzz** [comment]
> Also, a bit weird to attach functions to a React Component?

I think this is okay tbh. With a functional API, consumers could also do:

```jsx
const Button = styled('button', {

}).variant('variant', {
	muted: {}
}).variant('size', {
	small: {},
	large: {}
});
```

Which isn't too dissimilar to the following with `styled-components`:

```jsx
const Button = styled('button').attrs(() => ({ 
	type: 'button' 
}))({
  color: 'red',
});
```

I think this would be a nice addition to the API 🙂

**Comment by colmtuite** [comment]
Is something like this possible?

```
const Button = styled('button')(
  {
    backgroundColor: 'gray',
  },
  variants(
  {
    size: {
      small: {
        fontSize: '12px',
      },
      large: {
        fontSize: '16px'
      },
    },
    color: {
      pink: {
        backgroundColor: 'pink',
      },
      orange: {
        backgroundColor: 'orange'
      },
    },
  })
)
```

**Comment by colmtuite** [comment]
Nvm, @hadihallak has since informed me that I'm a dumbass.

**Comment by hadihallak** [comment]
Chatted with Colm about this and we're gonna go with this syntax for now while making sure it's typeable and then iterate from there if needed:
```
const Button = styled();
Button.variant('size', {
  small: {},
  large: {}
})
Button.variant('color', {
  small: {},
  large: {}
})
```

**Comment by peduarte** [comment]
Cool, SGMT

**Comment by hadihallak** [comment]
Resolved in #134
